virtual-desktop/src/app/plugin-manager/plugin-factory/iframe/iframe-plugin.component.ts

```diff
@@ -246,6 +251,14 @@ export class IFramePluginComponent {
             this.responses[args[0]].resolve();
           }
           return undefined;
+        case 'getPluginDefinition':
+          try{
+            return JSON.parse(JSON.stringify(this.pluginDefintion))
+          }catch (e){
+            return undefined;
+          }
+        case 'getLaunchMetadata':
```

Comment:
Don't pull, since the angular & react API is a push: when the constructor of the main component executes, the injectables are present whether or not they are used. Make the iframe API the same way in that when the iframe says it is ready, the desktop can post a message that contains all info necessary to create objects that are like those injectables. So, you need the iframe adapter to have some sort of event that the iframe code listens on to await this constructor response.

iframe --- READY ---> desktop
desktop ---- pluginDef, logger ID, ways to listen on window&viewport ----> iframe adapter
iframe adapter ---- event containing pluginDef, actual component logger, viewport & window objects ---> iframe code